src/wallet/wallet.cpp

```diff
@@ -1961,16 +1969,17 @@ std::optional<ZTXOSelector> CWallet::ZTXOSelectorForViewingKey(
     }, vk);
 
     if (pattern.has_value()) {
-        return ZTXOSelector(pattern.value(), requireSpendingKey);
+        return ZTXOSelector(pattern.value(), requireSpendingKey, requireTransparentCoinbase);
     } else {
         return std::nullopt;
     }
 }
 
-ZTXOSelector CWallet::LegacyTransparentZTXOSelector(bool requireSpendingKey) {
+ZTXOSelector CWallet::LegacyTransparentZTXOSelector(bool requireSpendingKey, bool coinbaseOnly) {
```

Comment:
I think there's some lack of clarity around the meaning of this flag, because where it's used I *think* it means "onlyTransparentCoinbase". The naming around this should be consistent and it should be documented.

Reply:
Fixed.